File: MojPrijevozFront/moj_prijevoz_admin/lib/resources/responses/rating/rating_response.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:moj_prijevoz_admin/resources/responses/fare/fare_response.dart';
import 'package:moj_prijevoz_admin/resources/responses/rating/all_ratings_response.dart';
import 'package:moj_prijevoz_admin/resources/responses/user_profile/user_profile_response.dart';

part 'rating_response.g.dart';

@JsonSerializable()
class RatingResponse extends AllRatingsResponse {
  final String? comment;
  final int fareId;
  final FareResponse? fare;
  RatingResponse({
    this.comment,
    required this.fareId,
    this.fare,
    required super.id,
    required super.fromId,
    required super.toId,
    required super.grade,
    required super.isVisible,
    required super.createdAt,
    super.from,
    super.to,
  });

  factory RatingResponse.fromJson(Map<String, dynamic> json) =>
      _$RatingResponseFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$RatingResponseToJson(this);
}

File: MojPrijevoz/MojPrijevoz.Model/Responses/Rating/RatingResponse.cs
using MojPrijevoz.Database;

namespace MojPrijevoz.Model.Responses.Rating;

public class RatingResponse
{
    public int Id { get; set; }

    public int FromId { get; set; }


    public string? Comment { get; set; }

    public short Grade { get; set; }

    public int FareId { get; set; }

    public UserProfile? From { get; set; }


    public DateTime CreatedAt { get; set; }
}